app/views/report_templates/jobs_invocations.erb

```diff
@@ -0,0 +1,35 @@
+<%#
+name: Jobs - Invocations
+snippet: false
+template_inputs:
+- name: job_id
+  required: true
+  input_type: user
+  description: ID of job to report
+  advanced: false
+  value_type: plain
+  resource_type: JobInvocation
+  hidden_value: false
+- name: hosts
+  required: false
+  input_type: user
+  description: Field for filter hosts of job invocation. Leave blank for all hosts.
+  advanced: false
+  value_type: search
+  resource_type: Host
+  hidden_value: false
+model: ReportTemplate
+-%>
+<%- invocation = load_job_invocation_by_id(input('job_id')) -%>
+<%- search_in_hosts_relation(invocation.template_invocations_hosts, input('hosts')).each do |host| -%>
```

Comment:
BTW, would this work instead of the core PR?
```suggestion
<%- load_hosts(search: input('hosts'), where: {id: template_invocations_hosts.pluck(:id)}).each do |host| -%>
```

Reply:
It should probably work. I understand that this makes core PR pointless but also I like to don't bother users with AR methods.  What do you think? What think others (@ares or @adamruzicka) ?

Reply:
I wouldn't be too concerned about users needing to understand AR here, there's very low chance they would be modifying this template. Also it may be easier for them to understand `load_hosts` which is well documented and used everywhere else. I'd be ok with the change, proven it works with safe mode. I'm not sure `pluck` is allowed.

Reply:
I understand that `.map(&:id)` is much more easier. On the other hand, `.map` method (`.pluck` is not allowed) loads whole object to memory. I think that the host is one of complex models in Foreman. That means that suggested code loads whole host model only if `id`. It produces longer time to generate the report for large numbers of hosts. 

I'm little worried about this. What do you think @ares or @ezr-ondrej ?

Reply:
Perhaps we should add `.ids` method to AR jails in foreman core?

Reply:
Great idea @tbrisker!